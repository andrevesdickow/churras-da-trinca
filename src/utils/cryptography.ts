import crypto from 'crypto';
import { isEmpty } from 'lodash';

function getBytes(str: string) {
  const bytes = [];
  for (let i = 0; i < str.length; ++i) {
    bytes.push(str.charCodeAt(i));
  }
  return bytes;
}

/**
 * Criptografa um texto
 * @param textToEncrypt texto
 * @param secretKey chave secreta
 * @returns texto criptografado
 */
export function encrypt(textToEncrypt: string, secretKey: string) {
  if (isEmpty(textToEncrypt) || isEmpty(secretKey)) {
    return '';
  }

  const key = Buffer.from(getBytes(secretKey));
  const iv = Buffer.from([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);

  const cipher = crypto.createCipheriv('aes-128-cbc', key, iv);
  let encrypted = cipher.update(textToEncrypt);
  encrypted = Buffer.concat([encrypted, cipher.final()]);
  return encrypted.toString('base64');
}
