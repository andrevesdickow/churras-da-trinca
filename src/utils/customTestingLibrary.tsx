import '../app/global.css';
import React from 'react';
import { ToastContainer } from '@/components/Toast';
import { NextThemeProvider } from '@/providers/NextThemeProvider';
import { Provider as TooltipProvider } from '@radix-ui/react-tooltip';
import { render } from '@testing-library/react';

const customRender = (ui: React.ReactElement, { ...renderOptions } = {}) => render(
  <NextThemeProvider>
    <TooltipProvider>
      {ui}
      <ToastContainer />
    </TooltipProvider>
  </NextThemeProvider>,
  renderOptions
);

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
