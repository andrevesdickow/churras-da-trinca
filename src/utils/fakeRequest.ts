export function fakeRequest<T = any>(delay: number, value: T) {
  return new Promise((resolve) => setTimeout(resolve, delay, value));
}
