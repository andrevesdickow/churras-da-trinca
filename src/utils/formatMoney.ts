/**
 * Formata valor monetÃ¡rio para real brasileiro
 * @param value valor
 * @returns Valor formatado -> R$ 1.000,00
 */
export function formatMoney(value: number) {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL'
  }).format(value);
}
