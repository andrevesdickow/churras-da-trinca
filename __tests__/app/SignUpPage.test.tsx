import SignUpPage from '@/app/signup/page';
import { fireEvent, render, screen } from '@/utils/customTestingLibrary';

const push = jest.fn();

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      push
    };
  }
}));

describe('<SignUpPage />', () => {
  it('should render correctly SignUp Page', () => {
    const { container } = render(<SignUpPage />);

    expect(screen.getByText(/Trinca/i)).toBeInTheDocument();
    expect(screen.getByText(/Cadastrar/i)).toBeInTheDocument();
    expect(screen.getByText(/Voltar para login/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should click button correctly submit form in SignUp Page', async () => {
    render(<SignUpPage />);

    fireEvent.submit(screen.getByRole('button', { name: 'Cadastrar' }));

    expect(await screen.findAllByRole('alert')).toHaveLength(5);
    expect(await screen.findByText('Nome é obrigatório')).toBeInTheDocument();
    expect(await screen.findByText('E-mail é obrigatório')).toBeInTheDocument();
    expect(await screen.findByText('Senha deve possuir no mínimo 8 caracteres')).toBeInTheDocument();
    expect(await screen.findByText('Confirmar senha deve possuir no mínimo 8 caracteres')).toBeInTheDocument();
    expect(await screen.findByText('Você precisa aceitar os termos')).toBeInTheDocument();

    fireEvent.input(screen.getByRole('textbox', { name: 'Nome completo' }), { target: { value: 'John Doe' } });
    fireEvent.input(screen.getByRole('textbox', { name: 'E-mail' }), { target: { value: 'john@doe.com' } });
    fireEvent.input(screen.getByRole('textbox', { name: 'Senha' }), { target: { value: '12345678' } });
    fireEvent.input(screen.getByRole('textbox', { name: 'Confirmar senha' }), { target: { value: '12345678' } });
    fireEvent.click(screen.getByRole('checkbox', { name: 'Confirmar e aceitar os termos de uso' }), { target: { value: 'true' } });
  });

  it('should click create barbecue link correctly in SignUp Page', () => {
    render(<SignUpPage />);

    expect(screen.getByRole('link', { name: 'Voltar para login' })).toHaveAttribute('href', '/');
  });
});
