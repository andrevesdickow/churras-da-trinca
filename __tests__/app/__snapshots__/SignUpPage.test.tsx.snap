// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage /> should render correctly SignUp Page 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <div
    class="w-full h-screen overflow-hidden flex justify-center items-center bg-amber-400 bg-barbecue"
  >
    <form
      class="flex flex-col gap-3 p-8 rounded-3xl z-[1px] min-w-[400px] max-w-[440px] h-max bg-slate-50 dark:bg-slate-800 shadow-xl"
    >
      <h3
        class="text-center font-semibold text-primary-900 dark:text-slate-50"
      >
        Faça seu cadastro para acompanhar os churras da 
        <span
          class="font-bold uppercase"
        >
          Trinca
        </span>
      </h3>
      <div>
        <div
          class="relative z-0 mb-4"
        >
          <input
            class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
            id="name"
            name="name"
            placeholder=" "
            role="textbox"
            type="text"
            value=""
          />
          <label
            class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
            for="name"
          >
            Nome completo
          </label>
        </div>
        <div
          class="relative z-0 mb-4"
        >
          <input
            class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
            id="email"
            name="email"
            placeholder=" "
            role="textbox"
            type="email"
            value=""
          />
          <label
            class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
            for="email"
          >
            E-mail
          </label>
        </div>
        <div
          class="relative z-0 mb-4"
        >
          <input
            class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
            id="password"
            name="password"
            placeholder=" "
            role="textbox"
            type="password"
            value=""
          />
          <label
            class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
            for="password"
          >
            Senha
          </label>
          <div
            class="absolute h-10 flex items-center right-1 top-0"
          >
            <button
              aria-label="Mostrar"
              class="inline-flex items-center justify-center gap-2 rounded-3xl text-sm font-medium uppercase ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 bg-transparent text-slate-950 hover:bg-amber-100 dark:bg-transparent dark:text-slate-300 dark:hover:bg-amber-400 dark:hover:text-slate-950 h-8 w-8"
              data-state="closed"
              type="button"
            >
              <svg
                class="w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 12s3-7 10-7 10 7 10 7-3 7-10 7-10-7-10-7Z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </button>
          </div>
          <div
            class="text-[0.625rem] mt-1 text-gray-500"
            role="note"
          >
            A senha deve possuir no mínimo 8 caracteres, uma letra maiúscula, uma letra minúscula, um número e um caractere especial
          </div>
          <div
            style="position: relative;"
          >
            <div
              style="display: flex; align-items: center; margin: 5px 0px 0px;"
            >
              <div
                style="flex-basis: 0px; flex-grow: 1; position: relative; max-width: 100%; width: 100%; height: 2px; background-color: rgb(221, 221, 221);"
              />
              <div
                style="width: 4px;"
              />
              <div
                style="flex-basis: 0px; flex-grow: 1; position: relative; max-width: 100%; width: 100%; height: 2px; background-color: rgb(221, 221, 221);"
              />
              <div
                style="width: 4px;"
              />
              <div
                style="flex-basis: 0px; flex-grow: 1; position: relative; max-width: 100%; width: 100%; height: 2px; background-color: rgb(221, 221, 221);"
              />
              <div
                style="width: 4px;"
              />
              <div
                style="flex-basis: 0px; flex-grow: 1; position: relative; max-width: 100%; width: 100%; height: 2px; background-color: rgb(221, 221, 221);"
              />
            </div>
            <p
              class="impo text-[0.625rem] text-slate-500"
              style="margin: 5px 0px 0px; color: rgb(137, 135, 146); font-size: 14px; text-align: right;"
            >
              curta demais
            </p>
          </div>
        </div>
        <div
          class="relative z-0 mb-4"
        >
          <input
            class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
            id="confirmPassword"
            name="confirmPassword"
            placeholder=" "
            role="textbox"
            type="password"
            value=""
          />
          <label
            class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
            for="confirmPassword"
          >
            Confirmar senha
          </label>
          <div
            class="absolute h-10 flex items-center right-1 top-0"
          >
            <button
              aria-label="Mostrar"
              class="inline-flex items-center justify-center gap-2 rounded-3xl text-sm font-medium uppercase ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 bg-transparent text-slate-950 hover:bg-amber-100 dark:bg-transparent dark:text-slate-300 dark:hover:bg-amber-400 dark:hover:text-slate-950 h-8 w-8"
              data-state="closed"
              type="button"
            >
              <svg
                class="w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 12s3-7 10-7 10 7 10 7-3 7-10 7-10-7-10-7Z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="flex flex-col mb-2"
        >
          <div
            class="flex gap-2 items-center"
          >
            <button
              aria-checked="false"
              class="peer h-4 w-4 shrink-0 rounded-sm border border-amber-300 ring-offset-white focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-amber-300 data-[state=checked]:text-white dark:border-amber-300 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 dark:data-[state=checked]:bg-amber-300 dark:data-[state=checked]:text-white"
              data-state="unchecked"
              id="confirmAndAgree"
              role="checkbox"
              type="button"
              value="false"
            />
            <input
              aria-hidden="true"
              name="confirmAndAgree"
              style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
              tabindex="-1"
              type="checkbox"
              value="false"
            />
            <label
              class="text-sm text-gray-500 dark:text-gray-300 hover:cursor-pointer"
              for="confirmAndAgree"
            >
              Confirmar e aceitar os termos de uso
            </label>
          </div>
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 rounded-3xl text-sm font-medium uppercase ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 bg-slate-950 text-white hover:bg-slate-950/90 dark:bg-slate-950 dark:text-white dark:hover:bg-slate-950/50 h-10 px-4 py-2"
        type="submit"
      >
        Cadastrar
      </button>
      <a
        class="text-amber-400 hover:text-amber-500 transition-colors text-center text-sm"
        href="/"
      >
        Voltar para login
      </a>
    </form>
  </div>
</div>
`;
